cibyl/orchestrator.py

```diff
@@ -66,14 +66,35 @@ def create_ci_environments(self) -> None:
         except AttributeError as exception:
             raise InvalidConfiguration from exception
 
+    def populate(environment, model_instances):
+        """Populate environment instance with the provided model instances"""
+        for model_instance in model_instances:
+            pass
+            # TODO: propose a method to populate the environment
+            #       with provided model instances
+
     def run_query(self, start_level=1):
         """Execute query based on provided arguments."""
         last_level = -1
         for arg in sorted(self.parser.ci_args.values(),
                           key=operator.attrgetter('level'), reverse=True):
             if arg.level >= start_level and arg.level >= last_level:
-                LOG.debug("executing the function %s", arg.func)
-                last_level = arg.level
+                for env in self.environments:
+                    for system in env.systems:
+                        valid_sources = []
+                        for source in system.sources:
+                            if hasattr(source, arg.func):
+                                valid_sources.append(source)
+                        if len(valid_sources) == 0:
+                            raise NoSupportedSourcesFound
+                        elif len(valid_sources) > 2:
+                            raise TooManyValidSources
+                        else:
```

Comment:
This else is not really needed, it's just a matter of style, but pylint seems to not like and I would agree